SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Tests/TestDefinitions.h"
+
+#include "SpatialWorkerConnection.h"
+
+#include "Interop/SpatialOutputDevice.h"
+
+#include "SpatialGDKServices/LocalDeploymentManager/LocalDeploymentManagerUtilities.h"
+
+#include "CoreMinimal.h"
+
+#define WORKERCONNECTION_TEST(TestName) \
+	GDK_TEST(Core, SpatialWorkerConnection, TestName)
+
+using namespace SpatialGDK;
+
+namespace
+{
+bool bClientConnectionProcessed = false;
+bool bServerConnectionProcessed = false;
+double MAX_WAIT_TIME = 10.0;
+
+void ConnectionProcessed(bool bConnectAsClient)
+{
+	if (bConnectAsClient)
+	{
+		bClientConnectionProcessed = true;
+	}
+	else
+	{
+		bServerConnectionProcessed = true;
+	}
+}
+
+void StartSetupConnectionConfigFromURL(USpatialWorkerConnection* Connection, const FURL& URL, bool& bOutUseReceptionist)
+{
+	bOutUseReceptionist = !(URL.Host == SpatialConstants::LOCATOR_HOST || URL.HasOption(TEXT("locator")));
```

Comment:
```suggestion
	bOutUseReceptionist = URL.Host != SpatialConstants::LOCATOR_HOST && !URL.HasOption(TEXT("locator"));
```
Nit: I find this easier to read. Also: don't we have similar logic elsewhere? Maybe we could reuse it.

Reply:
We do have the same code in SpatialNetDriver. Since it's a nit, I'll keep it as it is for now, so that it's the same as in SpatialNetDriver